src/store/modules/product/actions.ts

```diff
@@ -83,10 +83,21 @@ const actions: ActionTree<ProductState, RootState> = {
     commit(types.PRODUCT_ADD_TO_UPLD_PRDTS, { product: payload })
   },
 
-  updateCurrentProduct ({ commit, state }, payload) {
+  async updateCurrentProduct ({ commit, state }, payload) {
     // search in uploadProducts that if the clicked product is already in the upload list and set it as current product
-    const currentProduct = state.uploadProducts[payload.product.sku]
-    commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct ? currentProduct : payload.product })
+    const currentProduct = state.uploadProducts[payload]
+    if(currentProduct) {
+      commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct })
+    } else {
+      await ProductService.fetchProducts({
+        // used sku as we are currently only using sku to search for the product
+        "filters": ['sku: ' + '*' + payload + '*', 'isVirtual: false'],
+      }).then((resp)=>{
+        commit(types.PRODUCT_CURRENT_UPDATED, { product: resp.data.response.docs[0] })
```

Comment:
Handle the case when we have a response, but the status of the response is not 200, or the response contains error field, or the response has empty data.